src/modules/apps/e2e/report.tsx

```diff
@@ -0,0 +1,270 @@
+
+  const [timePassed, setTimePassed] = React.useState(requestedAt ? Math.floor((Date.now() - requestedAt) / 1000) : null)
+
+  React.useEffect(() => {
+    if (timePassed == null || running.length === 0) return
+    const timeout = setTimeout(() => setTimePassed(Math.floor((Date.now() - requestedAt) / 1000)), 1000)
+    return () => clearTimeout(timeout)
+  })
+
+  return (
+    <React.Fragment>
+      <Box flexDirection="column" marginRight={1}>
+        <Text bold>TestId:</Text>
+        <Text bold>Apps:</Text>
+        <Text bold>Specs:</Text>
+        {timePassed ? <Text bold>Time:</Text> : null}
+      </Box>
+      <Box flexDirection="column">
+        <Text>{testId}</Text>
+        <Text>
+          <Color greenBright>{pass} passed</Color>, {pass} of {pass + fail} total
+        </Text>
+        <Text>
+          <Color greenBright>{passSpecs} passed</Color>, {passSpecs} of {totalSpecs} total
+        </Text>
+        {timePassed ? <Text>{`${timePassed}s`}</Text> : null}
+      </Box>
+    </React.Fragment>
+  )
+}
+
+const Report: React.FunctionComponent<ReportProps> = ({ completed, running }) => (
+  <Box flexDirection="column">
+    <Static>
+      {completed.map(([appId, specs]) => (
+        <Completed key={appId} appId={appId} specs={specs} />
+      ))}
+    </Static>
+
+    {running.length > 0 && (
+      <Box flexDirection="column" marginTop={1}>
+        {running.map(([appId, specs]) => (
+          <Running key={appId} appId={appId} specs={specs} />
+        ))}
+      </Box>
+    )}
+  </Box>
+)
+
+type RealTimeReport = {
+  testId: string
+  poll: () => Promise<TestReport>
+  interval: number
+  initialReport: TestReport
+  requestedAt?: number
+}
+
+const parseReport = (report: TestReport): ReportProps => {
+  const { completed = [], running = [] } = groupBy(
+    ([, specs]) => (completedApp(specs) ? 'completed' : 'running'),
+    toPairs(report)
+  )
+  return { completed, running }
+}
+
+export const RealTimeReport: React.FunctionComponent<RealTimeReport> = ({
+  testId,
+  poll,
+  interval,
+  initialReport,
+  requestedAt,
+}) => {
+  const [delay, setDelay] = React.useState(interval)
+  const [report, setReport] = React.useState<ReportProps>(parseReport(initialReport))
+
+  const handleReport = ({ completed, running }: ReportProps) => {
+    const recentlyCompleted = difference(completed, report.completed)
+
+    setReport({
+      // careful not to reorder completed apps because they are Static
+      completed: [...report.completed, ...recentlyCompleted],
+      running,
+    })
+
+    if (running.length === 0) setDelay(null)
+  }
+
+  useInterval(
+    () =>
+      poll()
+        .then(parseReport)
+        .then(handleReport),
+    delay
+  )
+
+  return (
+    <Box flexDirection="column">
+      <Report {...report} />
+      <Box marginTop={1}>
+        <Summary {...report} testId={testId} requestedAt={requestedAt} />
+      </Box>
+    </Box>
+  )
+}
```

Comment:
How about something like:
```ts
const Report: React.FunctionComponent<ReportProps> = ({ completedAppTests, runningAppTests }) => (
  <Box flexDirection="column">
    <Static>
      {completedAppTests.map(({ appId, specs }) => (
        <Completed key={appId} appId={appId} specs={specs} />
      ))}
    </Static>

    {runningAppTests.length > 0 && (
      <Box flexDirection="column" marginTop={1}>
        {runningAppTests.map(({ appId, specs }) => (
          <Running key={appId} appId={appId} specs={specs} />
        ))}
      </Box>
    )}
  </Box>
)

interface RealTimeReport {
  testId: string
  poll: () => Promise<TestReport>
  interval: number
  initialReport: TestReport
  requestedAt?: number
}

interface AppTest {
  appId: string
  specs: AppReport
}

interface ReportProps {
  completedAppTests: AppTest[]
  runningAppTests: AppTest[]
}

const parseReport = (report: TestReport): ReportProps => {
  const appTests = Object.keys(report).map((appId) => {
    return { 
      appId, 
      specs: report[appId] 
    }
  })

  const completedAppTests = appTests.filter((appTest) => completedApp(appTest.specs))
  const runningAppTests = appTests.filter((appTest) => !completedApp(appTest.specs))
  return { completedAppTests, runningAppTests }
}

export const RealTimeReport: React.FunctionComponent<RealTimeReport> = ({
  testId,
  poll,
  interval,
  initialReport,
  requestedAt,
}) => {
  const [delay, setDelay] = React.useState(interval)
  const [report, setReport] = React.useState<ReportProps>(parseReport(initialReport))

  const handleReport = ({ completedAppTests, runningAppTests }: ReportProps) => {
    const recentlyCompleted = difference(completedAppTests, report.completedAppTests)

    setReport({
      // careful not to reorder completed apps because they are Static
      completedAppTests: [...report.completedAppTests, ...recentlyCompleted],
      runningAppTests,
    })

    if (runningAppTests.length === 0) setDelay(null)
  }

  useInterval(
    () =>
      poll()
        .then(parseReport)
        .then(handleReport),
    delay
  )

  return (
    <Box flexDirection="column">
      <Report {...report} />
      <Box marginTop={1}>
        <Summary {...report} testId={testId} requestedAt={requestedAt} />
      </Box>
    </Box>
  )
}
```
- Changed some types namings
- Reduced ramda usage
- Made variables names more hintable